src/ImageRunModal.jsx

```diff
@@ -58,6 +153,10 @@ export class ImageRunModal extends React.Component {
             createConfig.resources.memory = this.state.memory * (1024 ** units[this.state.memoryUnit].base1024Exponent);
         if (this.state.hasTTY)
             createConfig.tty = true;
+        if (this.state.publish.length > 0)
+            createConfig.publish = this.state.publish
+                    .filter(port => port.hostPort && port.containerPort)
```

Comment:
I verified that 0 is not a valid port number, otherwise this would be wrong (you'd need to check for `!== null`).